core/controllers/admin.py

```diff
@@ -204,6 +210,43 @@ def _reload_collection(self, collection_id):
         else:
             raise Exception('Cannot reload a collection in production.')
 
+    def _generate_dummy_exploration(self, dummy_exp_count, dummy_exp_publish):
+        """
+        Generates and Publishes given number of dummy explorations.
+
+        Args:
+            dummy_exp_count: int. Count of dummy explorations to be generated.
+            dummy_exp_publish: int. Count of explorations to be published.
+
+        Raises:
+            Exception: Environment is not DEVMODE.
+        """
+
+        if feconf.DEV_MODE:
+            logging.info(
+                '[ADMIN] %s generated %s number of dummy explorations' %
+                (self.user_id, dummy_exp_count))
+            possible_titles = ['Hulk Neuroscience', 'Quantum Starks',
+                               'Wonder Anatomy',
+                               'Elvish, language of "Lord of the Rings',
+                               'The Science of Superheroes']
+            dummy_exp_list = []
+            for _ in range(dummy_exp_count):
+                title = random.choice(possible_titles)
+                category = random.choice(feconf.SEARCH_DROPDOWN_CATEGORIES)
+                new_exploration_id = exp_services.get_new_exploration_id()
+                exploration = exp_domain.Exploration.create_default_exploration(
+                    new_exploration_id, title=title, category=category,
+                    objective='Dummy Objective')
+                exp_services.save_new_exploration(self.user_id, exploration)
+                dummy_exp_list.append(new_exploration_id)
+                if _ <= (dummy_exp_publish - 1):
+                    rights_manager.publish_exploration(
+                        self.user_id, new_exploration_id)
+            exp_services.index_explorations_given_ids(dummy_exp_list)
```

Comment:
I'm a bit confused about why this is needed. Isn't it already called during save_exploration()?